lib/Mojo/WebSocketProxy/Backend/JobAsync.pm

```diff
@@ -120,11 +123,14 @@ sub call_rpc {
     my $params = $self->make_call_params($c, $req_storage);
     $log->debugf("method %s has params = %s", $method, $params);
     $_->($c, $req_storage) for @$before_call_hook;
-    $self->client->submit(
-        name   => $req_storage->{name},
-        params => encode_json_utf8($params)
-        )->on_ready(
-        sub {
+
+    Future->wait_any(
+        $self->client->submit(
+            name   => $req_storage->{name},
+            params => encode_json_utf8($params)
+        ),
+        Future::Mojo->new_timer(RPC_QUEUE_TIMEOUT)->then(sub { Future->fail('Timeout') })
```

Comment:
why not `$self->loop->timeout_future(after => QUEUE_TIMEOUT)->on_fail(sub { Future->fail('timeout') })

Reply:
Tests were failing with `$self->{loop} undefined` error. I refactored the code a little bit by adding a new `$self->loop` and it was fixed with your suggested change.